### PR TITLE
fixed some clippy warnings in libcollections

### PR DESCRIPTION
**Pull request by llogiq**
Some markdown fixes, a few readability improvements, but mostly trivial stuff.

### CODE REVIEW

**Comment by alexcrichton** [comment]
@bors: retry

* ld.exe failure - https://github.com/rust-lang/rust/issues/40906

**Comment by alexcrichton** [comment]
@bors: retry

* ld.exe failure - https://github.com/rust-lang/rust/issues/40906

**Comment by alexcrichton** [comment]
@bors: retry

* ld.exe failure - https://github.com/rust-lang/rust/issues/40906

File: src/libcollections/binary_heap.rs [hunk 1]
```diff
@@ -1194,8 +1194,8 @@ impl<T: Ord, I: IntoIterator<Item = T>> SpecExtend<I> for BinaryHeap<T> {
 }
 
 impl<T: Ord> SpecExtend<BinaryHeap<T>> for BinaryHeap<T> {
-    fn spec_extend(&mut self, ref mut other: BinaryHeap<T>) {
-        self.append(other);
+    fn spec_extend(&mut self, mut other: BinaryHeap<T>) {
```
Comment (tamird) [inline src/libcollections/binary_heap.rs]: shouldn't this be `&mut other: BinaryHeap<T>`? I think you're forcing a copy here.

File: src/libcollections/linked_list.rs [hunk 2]
```diff
@@ -1020,8 +1020,8 @@ impl<I: IntoIterator> SpecExtend<I> for LinkedList<I::Item> {
 }
 
 impl<T> SpecExtend<LinkedList<T>> for LinkedList<T> {
-    fn spec_extend(&mut self, ref mut other: LinkedList<T>) {
-        self.append(other);
+    fn spec_extend(&mut self, mut other: LinkedList<T>) {
```
Comment (tamird) [inline src/libcollections/linked_list.rs]: ditto

**Comment by frewsxcv** [comment]
Going to pull this off the queue until @tamird's concerns are addressed

@bors r-

File: src/libcollections/binary_heap.rs (hunk 1 above)
Comment (bluss) [inline src/libcollections/binary_heap.rs]: This is purely a coding style issue and both versions are equivalent. Rust doesn't implicitly copy things unless they are Copy, and BinaryHeap is not and we can tell from the trait bounds that T doesn't have to be copy/clone at all to use this method.

File: src/libcollections/linked_list.rs (hunk 2 above)
Comment (bluss) [inline src/libcollections/linked_list.rs]: Rust doesn't implicitly copy things unless they are Copy, and heap-allocating collections are never Copy. Typical types that implement Copy are `i32` and `&str`.

**Comment by bluss** [comment]
@bors r+

It's fine as it is!

File: src/libcollections/binary_heap.rs (hunk 1 above)
Comment (bluss) [inline src/libcollections/binary_heap.rs]: I can further clarify that yes, the BinaryHeap is consumed / moved into the method here. In the old code and new code. It's ultimately dictated by the Extend interface and not something that can be changed.

**Comment by tamird** [comment]
👍

**Comment by alexcrichton** [comment]
@bors: retry

* ar.exe error https://github.com/rust-lang/rust/issues/40546

**Comment by alexcrichton** [comment]
@bors: retry

* ld.exe failed - https://github.com/rust-lang/rust/issues/40906

**Comment by frewsxcv** [comment]
@bors: retry

* ld.exe failed - https://github.com/rust-lang/rust/issues/40906

**Comment by frewsxcv** [comment]
@bors: retry

* ld.exe failed - https://github.com/rust-lang/rust/issues/40906

**Comment by alexcrichton** [comment]
@bors: retry

* network error https://github.com/rust-lang/rust/issues/40474

**Comment by alexcrichton** [comment]
@bors: retry

* ld.exe failure https://github.com/rust-lang/rust/issues/40906

**Comment by alexcrichton** [comment]
@bors: retry

* ld.exe error https://github.com/rust-lang/rust/issues/40906

**Comment by alexcrichton** [comment]
@bors: r-


er, hang on

**Comment by alexcrichton** [comment]
This PR is *always* failing with  https://github.com/rust-lang/rust/issues/40906, which makes me think it's not a spurious error. Turns out that's not happening on any other PRs as well. Even if this is just a bug in the MinGW linker, I don't think we can land this PR as-is.

**Comment by bluss** [comment]
Scouring through, I can't find anything suspicious in the diff.

**Comment by llogiq** [comment]
I think I may want to bisect this. Pity that travis doesn't show the error. Hang on...

**Comment by bluss** [comment]
Something hiding behind the merge in this branch?

**Comment by llogiq** [comment]
I'm closing this in favor of a series of smaller PRs.

**Pull request closed without merging**